### PR TITLE
Sb 94 Always send Heartbeat data

### PR DESCRIPTION
**Pull request by robester0403**
What does this code do?
Initiates send heartbeat at intropage, which is right now after login. Will continue to send heartbeats as long as the page is nested. Includes the fixes to match with server side which was not merged in time for this branch but it runs on its own.

What are potential issues?
The heartbeat are still not tracked by React Query and the errors are still handled in a console log way. The only question is how to demonstrate a loading state. Also how do we render and code error states since this can also be handled by react query.

Since we removed user roles the intro page requires small adjustment to switch from rider and driver. This is unavoidable until we can get login and user profiles set up.

### CODE REVIEW

File: src/pages/RouteDetailsRider/RouteDetailsRider.jsx [hunk 1]
```diff
@@ -59,16 +40,19 @@ const RouteDetailsRider = () => {
...
+          lng: heartBeatData.longitude || KINGSTON_JAMAICA_LONGITUDE,
+        }}
         containerWidth={width}
         containerHeight={height}
       >
-        {!isEmpty(riderGeoData) && (
+        {!isEmpty(heartBeatData) && (
           <MarkerF
-            key={riderGeoData.heartbeat_id + riderGeoData.time_stamp}
+            key={heartBeatData.userId + heartBeatData.userId}
```
Comment (yhus) [inline src/pages/RouteDetailsRider/RouteDetailsRider.jsx:52]: Is this a typo?

**Review by yhus — APPROVED** [review]
This is awesome!

To confirm, this is tested to work with the current heartbeat lambda service right (and not just a locally running instance)?

File: src/pages/RouteDetailsRider/RouteDetailsRider.jsx (hunk 1 above)
Comment (robester0403) [reply src/pages/RouteDetailsRider/RouteDetailsRider.jsx:75]: No from what I understand the names of the variables changed when we moved to AWS.

**Review by FioraDeMarco — APPROVED** [review]
Good work! We will refactor the routeDetailsDriver to do the same

**Pull request merged**